woocommerce/woocommerce
# CYS Redirect to intro screen if theme is not modified

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR introduces new states to the `assemblerHub` to check if the theme has been modified and redirect to the intro page if it has not.

Closes https://github.com/woocommerce/woocommerce/issues/43486



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Make sure to `Reset Customize Your Store` on `wp-admin/tools.php?page=woocommerce-admin-test-helper` if the store is now newly created.

1. Go to `wp-admin/admin.php?page=wc-admin&path=/customize-store/assembler-hub`.
2. Check it redirects to the intro screen (should not show the assembler).
3. Go to `WooCommerce > Home`, click on the `Start Customizing` button and finish the flow.
4. Go to `wp-admin/admin.php?page=wc-admin&path=/customize-store/assembler-hub`.
5. Check this time it does not redirect (should show the assembler).
6. Repeat 1-5 in a WooExpress store.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Redirect to the CYS intro screen when accessing the assembled hub without going through the customizing process.

#### Comment 

</details>
